content/blog/2019/06/2019-06-06-cloudfeature-gsoc-cp1.adoc

```diff
@@ -0,0 +1,104 @@
+---
+layout: post
+title: "GSoC Project Progress : Cloud features for external workspace manager"
+tags:
+- jenkins
+- gsoc
+author: imaffe
+description: "Short summary of coding period 1" # optional
+opengraph:
+  image: /images/folder/icon.png # optional
+---
+
+=== About Me
+My name is Yufei Zhang, a student from Zhejiang University and Simon Fraser University Duel Degree Program. I want to share the discoveries when we are exploring the possiblities of extending current external workspace manager plugin (referred to as EWM) to support AWS EFS. 
+
+=== Backgrounds
+For users, they might want to provision a high-available Jenkins cluster, together with other services such as Keycloak for SSO, and use k8s to autoscale agents. All these operations needs a lot of configuration and cloud resources if they decide to base Jenkins on clouds. 
+
+In such cases, EWM was designed to give multiple agents access to shared disks, hence reduce the cost of copying duplicated data, such as codes and outputs in the workspace, or archieves that needs to be managed by other services(e.g. Some companies might want use archieves to do some analysising). With these backgrounds, it is natural to think of "supporting cloud disks". 
+
+Our project proposal was based on extending cloud features, but then we discovered we were on the wrong track maybe.
+
+=== Discoveries
+In order to make things clear, we need to understand the true nature of EWM. As we stated earlier, we want multiple agents to share disk spaces, and the main obstacle was they all have their own workspace directory(paths) by default. If we can modify the default workspace directory, then we might be able to utilize the shared NFS(Network File System).  And that is what EWM did. There are other features in like different strategies choosing a disk, but the way EWM to solve the problem is simple but effective.
```

Comment:
Make sure you put a space between words here: "directory(paths)" -> "directory (paths)" and here "NFS(Network File System)" -> "NFS (Network ..)"